internal/api/token_refresh.go

```diff
@@ -80,6 +81,12 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 			if !notAfter.IsZero() && a.Now().After(notAfter) {
 				return oauthError("invalid_grant", "Invalid Refresh Token: Session Expired")
 			}
+
+			if config.Sessions.InactivityTimeout != nil {
+				if a.Now().Sub(session.LastRefreshedAt()) >= *config.Sessions.InactivityTimeout {
+					return oauthError("invalid_grant", "Invalid Refresh Token: Session Expired (Inactivity)")
+				}
+			}
```

Comment:
Yes, cleanup for this will happen with the regular cleanup as this way the error that the session expired remains visible to the user for a bit longer than if the session was destroyed immediately -- making a better DX.